mobile/examples/phi-3/android/app/src/main/java/ai/onnxruntime/genai/demo/MainActivity.java

```diff
@@ -84,9 +108,44 @@ public void onClick(View v) {
                 new Thread(new Runnable() {
                     @Override
                     public void run() {
+                        TokenizerStream stream = null;
+                        GeneratorParams generatorParams = null;
+                        Generator generator = null;
+                        Sequences encodedPrompt = null;
                         try {
-                            genAIWrapper.run(promptQuestion_formatted);
-                        } catch (GenAIException e) {
+                            stream = tokenizer.createStream();
+
+                            generatorParams = model.createGeneratorParams();
+                            //examples for optional parameters to format AI response
+                            //generatorParams.setSearchOption("length_penalty", 1000);
+                            //generatorParams.setSearchOption("max_length", 500);
+
+                            encodedPrompt = tokenizer.encode(promptQuestion_formatted);
+                            generatorParams.setInput(encodedPrompt);
+
+                            generator = new Generator(model, generatorParams);
```

Comment:
Can we do all this in the `try` so the close is automatically handled?

e.g. 
```java
try (stream = tokenizer.createStream();
     generatorParams = model.createGeneratorParams();
     ....) {
```

Reply:
I thought close would be automatically handled anyway since it's AutoClosable, based on what I read online. But if you think it's better to do it this way, please make the change.

Reply:
I believe `close()` is only automatically called when using a try-with-resources block (that construct shown in Scott's comment).
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/AutoCloseable.html